src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -1618,6 +1624,8 @@ type Money {
 
 """"
 An ISO-8601-formatted duration, i.e. `PT2H30M` for 2 hours and 30 minutes.
+
+Negative durations are formatted like `PT-10M`.
 """
 scalar Duration
```

Comment:
I think it would be better to have a Duration that only allows positive values and one that allows also negative values since in many use cases negative durations are not allowed. However, it would have been nicer to name this PositiveDuration for example instead of having to name a duration that also allows negative durations something specific. We can't rename this anymore since it is already being used. I think the best solution is to use this generic duration for that one use case that we already use this for (even though negative values don't make sense in that context) and I can add a PositiveDuration in the scope of https://github.com/opentripplanner/OpenTripPlanner/pull/5185 for new use cases.

Reply:
Can you remind me what we agreed in the meeting?

Reply:
https://github.com/opentripplanner/OpenTripPlanner/pull/5660/commits/47d28f1c8380efddad0871e58798313434b36f77

Reply:
We discussed this in the OTP meeting, and there is a borderline where you want to make new Scalars. Having one,  two or three durations might be ok, but not more than that. For example: `Duration`(no restriction), `SmallPosDuration`(Not negative, days, hours, minutes and seconds allowed only) - very common use-case in OTP. But, adding more restrictions into the type is probably not a good idea.

For negative durations I would only allow to have the minus sign in front of the duration (`-P1dT3s`), the spec allows `-P1dT3h-2m` - witch is not human readable.